sopel/modules/reddit.py

```diff
@@ -249,9 +249,54 @@ def comment_info(bot, trigger, match):
     bot.say(message)
 
 
+def subreddit_sorting(bot, trigger, s, sorting):
+    if sorting == 'new':
+        submissions = list(s.new(limit=10))
+    elif sorting == 'top':
+        submissions = list(s.top(limit=10))
+    elif sorting == 'hot':
+        submissions = list(s.hot(limit=10))
+    elif sorting == 'controversial':
+        submissions = list(s.controversial(limit=10))
+    elif sorting == 'gilded':
+        submissions = list(s.gilded(limit=10))
+    elif sorting == 'rising':
+        submissions = list(s.rising(limit=10))
+    elif sorting == 'sticky':
+        try:
+            submissions = [s.sticky()]
+        except prawcore.exceptions.NotFound:
+            bot.say("r/" + s.display_name + " appears to not have a stickied post!")
+            return
+    elif sorting == 'random':
+        submissions = [s.random()] or []
+    else:
+        raise ValueError("Unknown sorting type '%s'" % sorting)
+
+    if not len(submissions):
+        bot.say("r/" + s.display_name + ' ' + sorting + " appears to have no items!")
+        return plugin.NOLIMIT
+
+    if sorting != 'sticky':
+        submissions = [item for item in submissions if not item.stickied]
+    submission = submissions[0]
+
+    link = "https://reddit.com/r/" + s.display_name + "/comments/" + submission.id
+    say_post_info(
+        bot, trigger, re.match(post_url, link).group(1), False, True)
```

Comment:
Am I missing some blatantly obvious reason that building a URL just to parse it with the `post_url` regex is required? Can't we do this the easy way? The third parameter of `say_post_info()` is `id_`, the submission ID.

Plus, is there a reason not to show the URL for a post fetched this way? I'd say both the submitted URL _and_ the comments link are useful.

```suggestion
    say_post_info(bot, trigger, submission.id, True, True)
```